docs/tech-design/safety-pool.md

```diff
@@ -0,0 +1,86 @@
+# Safety Pool functionality for the Staking Pool
+
+| Specification | Implementation | Last revision |
+|:-----------:|:-----------:|:-------------:|
+| WIP         |  WIP        | 2022-05-13    |
+
+---
+
+**Specification ownership:** [Emily Martins]
+
+**Authors**:
+
+-   [Emily Martins]
+
+**Implementation ownership:** [Emily Martins]
+
+[Emily Martins]: https://github.com/emiflake
+
+[Jack Hodgkinson]: https://github.com/jhodgdev
+
+**Current status:** Initial draft in need of editing and review from [Jack Hodgkinson].
+
+---
+
+In order for Agora’s StakingPool to act as a *SafetyPool,* it needs to be able to support a workflow for slashing staked governance tokens to act as a safety mechanism. This document outlines the changes that Agora needs to receive in order to support this.
+
+### Motivation
+
+In the event of a protocol suffering loss of funds through a [shortfall event](https://docs.aave.com/aavenomics/safety-module#shortfall-events), slashing a percentage of locked GT can be used to attempt recovery. Ultimately, doing this is beneficial for the stakeholders because it allows the protocol to recover and eventually benefits them as well (even though they bear the initial cost). Striking a balance (in the form of the right % slashed) is important in order for the stakeholders to be willing to partake in their sacrifice.
+
+## Slashing functionality
+
+In order to allow an admin to withdraw a set percentage of the amount staked, we create a new effect.
+
+### The `SlashEffect` validator:
+
+- Mint a `SlashToken` and send it to a validator ("the `Slash` validator") with a datum encoding the details of the slashing.
+
+  The `SlashDatum` may look like this:
+
+  ```haskell
+  data SlashDatum = SlashDatum
+    { -- | Identify which slash event this datum belongs to.
+      slashId :: Integer
+    , -- | Represents how much is to be slashed (as a ratio of the full staked amount).
+      slashPercentage :: Rational
+    , -- | During what time is using this datum for Stakes.
+      slashTimeRange :: POSIXTimeRange
+    }
+  ```
+
+- This information must be encoded in the datum the effect is given to, so ultimately encoded in the proposal.
+  The datum that is passed to the effect ought to contain this `SlashDatum` in some way.
```

Comment:
Consider 'should' instead of 'ought to'.

P.S. this hurts as 'ought' is one of my favourite words in English. Especially 'ought not' shortened to 'oughtn't' :heart: